maintenance/wikia/AddDiscussionsToNavigation.php

```diff
@@ -0,0 +1,152 @@
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'verbose', 'Show extra debugging output', false, false, 'v' );
+		$this->addOption( 'rollback', 'Rollback the operation', false, false, 'r' );
+	}
+
+	public function execute() {
+		$this->dryRun = $this->hasOption( 'dry-run' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->rollback = $this->hasOption( 'rollback' );
+
+
+		$total = 0;
+		$successful = 0;
+		foreach ($this->getAllSiteIds() as $siteId) {
+			$success = $this->updateSiteId(trim($siteId));
+			if ($success) {
+				$successful++;
+			}
+			$total++;
+		}
+
+		echo "\nTotal sites: $total, Success: {$successful}, Failed: " . ($total - $successful) . "\n\n";
+	}
+
+	private function getAllSiteIds() {
+		$id = WikiFactory::getVarIdByName( self::VAR_NAME );
+		$db = WikiFactory::db( DB_MASTER );
+		$sql = ( new WikiaSQL() )
+			->SELECT( 'cv_city_id' )
+			->FROM( 'city_variables' )
+			->WHERE( 'cv_variable_id' )->EQUAL_TO( $id );
+
+		$siteIds = $sql->runLoop( $db, function( &$siteIds, $row ) {
+			$siteIds[] = $row->cv_city_id;
+			$this->debug("Added {$row->cv_city_id}");
+		});
+
+		return $siteIds;
+	}
+
+	private function updateSiteId($siteId) {
+		$varData = (array)WikiFactory::getVarByName( self::VAR_NAME, $siteId, true );
+		if ( empty( $varData['cv_id'] ) ) {
+			$this->debug("Error:" . self::VAR_NAME . " not found.\n");
+			return false;
+		}
+
+		if ( $varData['cv_variable_type'] !== 'text' ) {
+			echo $varData['cv_variable_type'];
+			die("Error: " . self::VAR_NAME . " is not a text and the script cannot append to it anything.\n");
+		}
+
+		$this->debug("Variable: " . self::VAR_NAME . " (Id: $varData[cv_id])\n");
+		$this->debug( "Variable data: " . json_encode( $varData ) . "\n");
+		$value = $this->createNewValue( unserialize( $varData[ 'cv_value' ] ) );
+
+		$result = false;
+		if ( !$this->dryRun ) {
+			$this->setVariable( $siteId, $value );
+		}
+		return $result;
+	}
+
+	private function createNewValue( $previousValue ) {
+		$this->debug( "Previous value:\n\n" . $previousValue );
+		$array = explode( self::SPLIT, $previousValue );
+		if ( !$this->rollback ) {
+			$array = $this->insertBeforeForumOrAtTheEnd( $array );
+		} else {
+			$array = $this->removeNavigationElement( $array );
+		}
+		$newValue = join( self::SPLIT, $array );
+		$this->debug( "New value\n\n:" . $newValue );
+		return $newValue;
+	}
+
+	function insertBeforeForumOrAtTheEnd(&$array ) {
+		$pos = array_search( self::FORUM_NAVIGATION_ELEMENT, $array );
+		if ( $pos === false ) {
+			$pos = count( $array ) - 1;
+			$this->debug( "Forum not found - appending at the end.\n" );
+		} else {
+			$this->debug( "Found forum at $pos position.\n");
+		}
+		return array_merge(
+			array_slice( $array, 0, $pos ),
+			[ self::NAVIGATION_ELEMENT ],
+			array_slice( $array, $pos )
+		);
+	}
+
+	function removeNavigationElement( &$array ) {
+		$pos = array_search( self::NAVIGATION_ELEMENT, $array );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)